### PR TITLE
housekeeping: Add Basic GitHub Actions Configuration

### PR DESCRIPTION
**Pull request by worldbeater**
This PR adds a basic GitHub Actions configuration which builds the project and runs the tests. Closes https://github.com/worldbeater/XamlNameReferenceGenerator/issues/1
Based on CI build scripts from the ReactiveUI.Validation repository https://github.com/reactiveui/ReactiveUI.Validation

### CODE REVIEW

**Comment by worldbeater** [comment]
Looks like we are unable to build the project via `dotnet build` because the sources aren't generated by the generator.

**Comment by worldbeater** [comment]
The release process now works like this: we open a PR, we add a `release` label, and once we merge the PR, a new version of the `XamlNameReferenceGenerator` ships to NuGet. This is how the flow should work at least, probably worth trying to ship the first version after this PR is merged. Also with this PR, we are using [Nerdbank.GitVersioning](https://github.com/dotnet/Nerdbank.GitVersioning) for incrementing assembly file versions.

File: src/XamlNameReferenceGenerator/XamlNameReferenceGenerator.csproj [hunk 1]
```diff
@@ -1,13 +1,22 @@
 <Project Sdk="Microsoft.NET.Sdk">
     <PropertyGroup>
         <TargetFramework>netstandard2.0</TargetFramework>
+        <Product>XamlNameReferenceGenerator ($(TargetFramework))</Product>
```
Comment (maxkatz6) [inline src/XamlNameReferenceGenerator/XamlNameReferenceGenerator.csproj]: Do we need specific TargetFramework here?

Comment (worldbeater) [reply src/XamlNameReferenceGenerator/XamlNameReferenceGenerator.csproj]: Good catch, don't remember why I decided to put it there. Removed.

**Pull request merged**